### PR TITLE
Fixes #24 - Introduce Macro usage to user documentation

### LINKED ISSUE #24: Documentation regarding Macro-Usuage
Right now the user documentation is missing the documentation of using Macros.
How to configure and use Macros can be found within the admin-documentation. At least the usuage part should move to user-docs.

### PR DESCRIPTION
**Pull request by MrGeneration**
This pull request introduces the Macro usage to the user documentation.
Prior this point it was indirectly covered by the admin documentation.

This PR depends on admin-doc PR https://github.com/zammad/zammad-admin-documentation/pull/93 .

---

This PR introduces a new file: `/advanced/macros.rst` that comes with fancy screencasts to support the text.
May be off in terms of wording, I tried to keep it short. :-)

Let me know if there's questions.

### CODE REVIEW

**Comment by rlue** [comment]
Here are my changes. I removed the following warning:

>     .. note:: ⚠ **Bulk operations come with a trade...**
>
>       Not for your soul, but some actions will not work. 
>       This affects the creation of articles and the "stay on tab" behavior. 

I don't think it's necessary to explain that "stay-on-tab" behavior doesn't apply to bulk macros. Have you had customers ask about that before?

And I'm not sure what you mean by "this affects the creation of articles." My understanding is that the only kinds of articles that macros can make are notes. Are you saying that when you run macros in bulk, the notes don't get created?

**Review by rlue — APPROVED** [review]
Approved with my latest QA commits, but (as always) open for further critique and discussion.

**Comment by MrGeneration** [comment]
> I don't think it's necessary to explain that "stay-on-tab" behavior doesn't apply to bulk macros. Have you had customers ask about that before?

This actually already brought confusion which is why I'd like to keep it.

> And I'm not sure what you mean by "this affects the creation of articles." My understanding is that the only kinds of articles that macros can make are notes. Are you saying that when you run macros in bulk, the notes don't get created?

I just tested this to be extra sure.
This statement is correct for Zammad 3.6 stable, but is not for develop.
We should keep it removed thus.

**Comment by rlue** [comment]
> This actually already brought confusion which is why I'd like to keep it.

🤔 Hm... The reason I wanted to remove it is that I think the warning belongs in the admin docs, not the user docs. Agents have no way of changing the "Once completed..." setting; in fact, they shouldn't even know about it. IMO this is a low-level detail that only admins should be concerned with. This setting isn't even mentioned in the user docs, and so explaining an edge case about its behavior is especially difficult/confusing.

When you had questions about it in the past, were they questions from users or admins?

**Comment by MrGeneration** [comment]
> Agents have no way of changing the "Once completed..." setting; in fact, they shouldn't even know about it

I disagree.
`Close, tag as spam and remove tab` macro names could tell what they do. In fact I did see such kind of macros already.
Also, the agent will learn quite fast what a macro does (it's relevant to know for the work the agent does).

So if the agent did learn that the macro in question does close the ticket tab if run via ticket detail view, it will be odd if it doesn't within bulk mode.

> When you had questions about it in the past, were they questions from users or admins?

I can't reliably tell, I'm sorry.
My guess is that both fractions have this question.

**Comment by rlue** [comment]
Actually, if you could wait another day on this one, I'd like to take this opportunity to reorganize some of the pages in the user docs.

The idea here is that agents/users should think at a different level of abstraction from admins. The **admin docs** introduce concepts strictly following Zammad's internal data model—there's an entry for triggers, the scheduler, macros, groups, and every other "object" in the system. This is appropriate for admins and expert users, but it's not necessarily how ideas should be organized in the **user docs**.

Agents may _use_ macros every day, but they never need to know that they're _called_ macros. The same is true of text modules: you can use them every day without knowing what they're called. To an agent, they're just `::` shortcuts. 

I'd like to rearrange things so that the ideas are more discoverable to a novice user (_i.e.,_ someone who has no idea what a macro or a text module is). I will keep the words "macro" and "text module" in the article titles, but I want to put related ideas closer together.

If you're not okay with this, you can go ahead and merge the PR branch as-is. If you _are_ okay with it, let me know, and I'll spend maybe two or three more hours applying these changes.

File: advanced/macros.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+1. open a ticket overview;
+2. select your desired tickets;
+3. click-and-drag to open the “Run Macro” drawer; and
+4. drop the tickets on your target macro.
+
+.. figure:: /images/advanced/macros/macro-usage-via-overview.gif
+   :width: 90%
+   :align: center
+   :alt: Screencast showing how to run macros via overviews.
```
Comment (rlue) [inline advanced/macros.rst:63]:
```suggestion
   :alt: Screencast showing how to run macros via overviews.
   
.. note:: ☝️ **There's just one difference...**

   When running a macro from the ticket view, Zammad may automatically open the next ticket (or close the current one, or just stay on it), depending on how the macro was set up.
   
   When running it from the overviews page, Zammad will always stay on the overviews page.
```

How's this?

Comment (MrGeneration) [reply advanced/macros.rst:63]: good once actually, I force push this one as I already merged. Thansk!

**Pull request merged**